File: src/Application Specific/Classes/Entity Collections/Van/EngineerVan/EngineerVanValidator.vb
Namespace Entity

    Namespace EngineerVans

        Public Class EngineerVanValidator
            'make sure that contact object is valid
            Inherits BaseValidator

            Public Sub Validate(ByVal oEngineerVan As EngineerVan)

                'make sure that contact object is valid
                If oEngineerVan.Errors.Count > 0 Then
                    Dim de As DictionaryEntry
                    For Each de In oEngineerVan.Errors
                        Me.AddCriticalMessage(CStr(de.Value))
                    Next
                End If

                If oEngineerVan.VanID = 0 Then
                    Me.AddCriticalMessage("Van Missing")
                End If
                If oEngineerVan.EngineerID = 0 Then
                    Me.AddCriticalMessage("Engineer Missing")
                End If
                If Not oEngineerVan.EndDateTime = Nothing Then
                    If Not oEngineerVan.StartDateTime < oEngineerVan.EndDateTime Then
                        Me.AddCriticalMessage("Start Date Time Must Be Before End Date Time")
                    End If
                End If

                If Me.ValidatorMessages.CriticalMessages.Count > 0 Then
                    Throw New ValidationException(Me)
                End If

            End Sub

        End Class

    End Namespace

End Namespace

File: apollo/apollo.fsm/src/Application Specific/Classes/Entity Collections/Van/EngineerVan/EngineerVanValidator.cs
using System.Collections;
using Microsoft.VisualBasic.CompilerServices;

namespace FSM.Entity
{
    namespace EngineerVans
    {
        // make sure that contact object is valid
        public class EngineerVanValidator : BaseValidator
        {
            public void Validate(EngineerVan oEngineerVan)
            {
                // make sure that contact object is valid
                if (oEngineerVan.Errors.Count > 0)
                {
                    foreach (DictionaryEntry de in oEngineerVan.Errors)
                        AddCriticalMessage(Conversions.ToString(de.Value));
                }

                if (oEngineerVan.VanID == 0)
                {
                    AddCriticalMessage("Van Missing");
                }

                if (oEngineerVan.EngineerID == 0)
                {
                    AddCriticalMessage("Engineer Missing");
                }

                if (oEngineerVan.EndDateTime != default)
                {
                    if (!(oEngineerVan.StartDateTime < oEngineerVan.EndDateTime))
                    {
                        AddCriticalMessage("Start Date Time Must Be Before End Date Time");
                    }
                }

                if (ValidatorMessages.CriticalMessages.Count > 0)
                {
                    throw new ValidationException(this);
                }
            }
        }
    }
}